lib/retry.ts

```diff
@@ -122,6 +137,11 @@ export async function retry<T>(config: RetryConfig<T>): Promise<T> {
       if (!err.translated) {
         err = translate(err);
       }
+      const isConnected = await checkNetworkConnection();
```

Comment:
Move the call to `checkNetworkConnection` inside the `if (!err.retryable && err.name === "ServiceCommunicationError")` to avoid making the dns call when it is not needed